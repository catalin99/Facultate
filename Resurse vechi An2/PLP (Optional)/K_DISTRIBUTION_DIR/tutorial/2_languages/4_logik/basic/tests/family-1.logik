parent(david,john).
parent(jim,david).
parent(steve,jim).
parent(nathan,steve).

grandparent(A,B) :-
  parent(A,X),
  parent(X,B).

ancestor(A,B) :-
  parent(A,B).
ancestor(A,B) :-
  parent(A,X),
  ancestor(X,B).

// This gives too many solutions, as it also includes A and B in solutions
?- ancestor(A,X), ancestor(X,B).
