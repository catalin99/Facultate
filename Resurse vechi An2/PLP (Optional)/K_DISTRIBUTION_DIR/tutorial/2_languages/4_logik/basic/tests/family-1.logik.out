Search results:

Solution 1:
Solution:Map -->
X:Term |-> david A:Term |-> jim B:Term |-> john

Solution 2:
Solution:Map -->
X:Term |-> david A:Term |-> nathan B:Term |-> john

Solution 3:
Solution:Map -->
X:Term |-> david A:Term |-> steve B:Term |-> john

Solution 4:
Solution:Map -->
X:Term |-> jim A:Term |-> nathan B:Term |-> david

Solution 5:
Solution:Map -->
X:Term |-> jim A:Term |-> nathan B:Term |-> john

Solution 6:
Solution:Map -->
X:Term |-> jim A:Term |-> steve B:Term |-> david

Solution 7:
Solution:Map -->
X:Term |-> jim A:Term |-> steve B:Term |-> john

Solution 8:
Solution:Map -->
X:Term |-> steve A:Term |-> nathan B:Term |-> david

Solution 9:
Solution:Map -->
X:Term |-> steve A:Term |-> nathan B:Term |-> jim

Solution 10:
Solution:Map -->
X:Term |-> steve A:Term |-> nathan B:Term |-> john

